### ISSUE TITLE
Bug: [no-unused-vars] False Positive for Unused Global Interfaces in Namespace-Based Projects

### ISSUE THREAD

**Issue by NoelAbrahams** [typescript-eslint/typescript-eslint#11028]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=LAKAlgdgLgpgTgMwIYGMYAIBiB7b6DeooAvkSBEgLYwDOADqhgEZJwFnrowAed2cUdCgA2SGjXQAhVujCU6wmNWgSceQiE4kyxIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUFs9xSAAgC4CeADigMbQCWZRAtCvDQHZED0rA9g7K7GSIAJgwBuAQ2ioMAbXB5F%2BAO5TWkADQKlSrDv2QpAcxlQJ8eJu37Fh6CYCSRntEQAFCUSKJo6jJAA9AH0rGx1ISglYIwALIgBRaGhuaXQzC1Cw20jouMTk6ScXd09vXzTAkK0s22EUIjhKIlhXYQBBJIkSIpSSrx8-KGDMmpxIGmdegCV6gGUaACMWVhM0hthEaxqAXy2bAF0FXe2gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjxYAEw7oKUIsSVSAviA1A&tokens=false

### Repro Code

```TypeScript
// File A.ts
interface Foo {

}


// File B.ts
namespace bar {

  export class Bar implements Foo {
  
}

}

// File C.ts
const baz = new bar.Bar();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
                "@typescript-eslint/no-unused-vars": [
                "warn",
                  {
                    "args": "all",
                    "argsIgnorePattern": "^_",
                    "caughtErrors": "all",
                    "caughtErrorsIgnorePattern": "^_",
                    "destructuredArrayIgnorePattern": "^_",
                    "ignoreRestSiblings": true
                  }
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "none"
  }
}
```

### Expected Result

No warnings or errors should be raised for:

- The Foo interface in A.ts, as it is implemented by the Bar class in B.ts.
- The bar namespace in B.ts, as it is used in C.ts.

### Actual Result

ESLint raises the following errors:

1. In A.ts:  'Foo' is defined but never used.
2. In B.ts: 'bar' is defined but never used.
 
These are false positives because both the Foo interface and the bar namespace are used globally in the project.

### Additional Info

When using @typescript-eslint in a TypeScript project structured around namespaces (without a module system), global interfaces and namespaces are flagged as unused by the @typescript-eslint/no-unused-vars rule, even though they are used elsewhere in the project. 

This appears to be a false positive, as both the interface and the namespace are intended to be globally available and are used in other parts of the codebase.

**Comment by bradzacher** [comment]
https://eslint.org/docs/latest/rules/no-unused-vars#exported

If you're not exporting via ESM you must a noted exported variables with comments as described in the docs.

**Comment by NoelAbrahams** [comment]
@bradzacher just tried this, but we don't allow inline configs (reason being it's just open to abuse).

So getting this:
`/* exported foo */' has no effect because you have 'noInlineConfig' setting in your config`

Appreciate the quick response, though.
Thanks.

**Comment by NoelAbrahams** [comment]
@bradzacher are you planning on catering for the scenario that I outlined in my last comment?

**Comment by bradzacher** [comment]
No.
The way to use the rule with this style of programming is to use comments annotating your "exported" variables.

If you don't allow such comments then your codebase is fundamentally incompatible with the rule.

**Comment by NoelAbrahams** [comment]
Sure. It looks like a unfixable bug in your library for a fairly simple use case.

**Comment by bradzacher** [comment]
Export comments have been the standard for declaring this for as long as this pattern has existed.

This isn't a pattern that we created. This is how eslint core has done things forever - well over a decade.

The fact that your codebase bans this well accepted pattern is your decision.

We have a solution, you've chosen not to use it. We will not be providing another solution. If this doesn't work for you, our code is open source, consider forking it to add support for your codebase's patterns.
